### PR TITLE
lib_pathbuffer: change nxmutex -> spinlock

### PR DESCRIPTION
**Pull request by crafcat7**
Summary:

## Summary

Change nxmutex to spinlock to fix bug:

```
_assert
|
 ->dump_fatal_info
   |
    ->dump_tasks
      |
       ->dump_filelist
         |
          ->files_dumplist
            |
             ->lib_get_pathbuffer
               |
                ->nxmutex_lock      <-- hold mutex on assert will trigger an nested exception
```

## Impact

**Is a new feature added?:** NO
**Impact on build:** NO
**Impact on hardware:** NO, this change does not specifically target any particular hardware architectures or boards.
**Impact on documentation:** NO,This patch does not introduce any new features
**Impact on compatibility:** This implementation aims to be backward compatible. No existing functionality is expected to be broken.

## Testing

Build Host(s): Linux x86
Test cases based on local qemu environment passed

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



Yes, this PR mostly meets the NuttX requirements, but could be improved.

Here's what's good:

* **Clear problem description:** The summary clearly identifies the bug and its root cause with a helpful stack trace.
* **Impact assessment:**  Addresses all impact points.
* **Testing information:** Provides host and target information, and mentions passing tests.

Here's what needs improvement:

* **Summary - "How does the change exactly work?" is missing:**  While the "why" is clear, the PR description lacks detail on *how* `nxmutex` is changed to a spinlock.  Explain the specific code changes made.  For example: "This PR replaces the `nxmutex_lock` call in `lib_get_pathbuffer` with a spinlock.  A new spinlock variable, `g_pathbuffer_lock`, is declared statically in the file and initialized during system startup."  Be precise.
* **Missing "Testing logs before/after change":**  While it mentions tests passing, it needs to show *actual* log output demonstrating the fix. Include relevant snippets demonstrating the error before the change and its absence after. Empty code blocks are insufficient.
* **Target details lacking:** "local qemu environment" is vague. Specify the QEMU architecture being used (e.g., `qemu-arm`, `qemu-riscv`). Mention the specific NuttX configuration (e.g., `nsh`).


By adding these details, the PR will be much stronger and easier to review.

**Comment by hujun260** [comment]
> ->nxmutex_lock      <-- hold mutex on assert will trigger an nested exception

Is this exception caused because assert is called within an interrupt?

**Comment by xiaoxiang781216** [comment]
> > ->nxmutex_lock      <-- hold mutex on assert will trigger an nested exception
> 
> Is this exception caused because assert is called within an interrupt?

get path is used in many places, so it's better to the interrupt safe.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by hujun260** [comment]
LGTM

File: libs/libc/misc/lib_pathbuffer.c [hunk 1]
```diff
@@ -25,6 +25,7 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
+#include <nuttx/spinlock.h>
 #include <nuttx/mutex.h>
```
Comment (anchao) [inline libs/libc/misc/lib_pathbuffer.c:29]:
```suggestion

```

File: libs/libc/misc/lib_pathbuffer.c [hunk 2]
```diff
@@ -124,21 +126,22 @@ FAR char *lib_get_pathbuffer(void)
...
       /* Mark the corresponding bit as free */
 
       g_pathbuffer.free_bitmap |= 1u << index;
-      nxmutex_unlock(&g_pathbuffer.lock);
+      spin_unlock_irqrestore(&g_pathbuffer.lock, flags);
       return;
     }
 
-  nxmutex_unlock(&g_pathbuffer.lock);
+  spin_unlock_irqrestore(&g_pathbuffer.lock, flags);
```
Comment (anchao) [inline libs/libc/misc/lib_pathbuffer.c:144]:
```suggestion
  index = (buffer - &g_pathbuffer.buffer[0][0]) / PATH_MAX;
  if (index >= 0 && index < CONFIG_LIBC_MAX_PATHBUFFER)
    {
      /* Mark the corresponding bit as free */
      flags = spin_lock_irqsave(&g_pathbuffer.lock);
      g_pathbuffer.free_bitmap |= 1u << index;
      spin_unlock_irqrestore(&g_pathbuffer.lock, flags);
      return;
    }
```

**Review by anchao — CHANGES_REQUESTED** [review]

File: libs/libc/misc/lib_pathbuffer.c (hunk 1 above)
Comment (crafcat7) [reply libs/libc/misc/lib_pathbuffer.c]: Done

File: libs/libc/misc/lib_pathbuffer.c (hunk 2 above)
Comment (crafcat7) [reply libs/libc/misc/lib_pathbuffer.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**